File: src/io/fdeitylink/keroedit/Messages.java
package io.fdeitylink.keroedit;

import java.util.ResourceBundle;
import java.util.MissingResourceException;

import io.fdeitylink.keroedit.util.Logger;

public final class Messages {
    private static final ResourceBundle resourceBundle = ResourceBundle.getBundle("io.fdeitylink.keroedit.messages");

    private Messages() {

    }

    /**
     * Returns a {@code String} that corresponds to a key in the {@code messages.properties}
     * resource bundle.
     *
     * @param key The key corresponding to the {@code String} to retrieve
     *
     * @return The {@code String} corresponding to {@code key}
     */
    public static String getString(final String key) {
        try {
            return resourceBundle.getString(key);
        }
        catch (final MissingResourceException except) {
            Logger.logThrowable("Missing string resource: " + key, except);
            return "Missing resource: " + key;
        }
    }
}

File: src/io/fdeitylink/keroedit/Messages.kt
package io.fdeitylink.keroedit

import java.util.ResourceBundle
import java.util.MissingResourceException

import io.fdeitylink.util.Logger

object Messages {
    private val resourceBundle = ResourceBundle.getBundle("io.fdeitylink.keroedit.messages")

    /**
     * Returns a [String] that corresponds to a key in the messages.properties
     * resource bundle in the [io.fdeitylink.keroedit] package.
     *
     * @param key the key corresponding to the [String] to retrieve
     *
     * @return the [String] corresponding to [key], or [key] if there is no
     * mapping for it in the resource bundle
     */
    operator fun get(key: String): String {
        try {
            return resourceBundle.getString(key)
        }
        catch (except: MissingResourceException) {
            Logger.logMessage("Missing string resource $key")
            return key
        }
    }
}